io-engine/src/lvs/lvol_snapshot.rs

```diff
@@ -0,0 +1,260 @@
+use crate::{
+    bdev::Nexus,
+    core::{CoreError, Mthread, VolumeSnapshot},
+    lvs::{lvs_lvol::LvsLvol, Lvol},
+    spdk_rs::ffihelper::IntoCString,
+    subsys::NvmfReq,
+};
+use async_trait::async_trait;
+use core::ffi::c_void;
+use spdk_rs::libspdk::{
+    spdk_bdev_io,
+    spdk_bdev_io_get_thread,
+    spdk_lvol,
+    spdk_nvmf_request_complete,
+    vbdev_lvol_create_snapshot,
+};
+/// LvolSnapshot Captures all the Snapshot information for Lvol
+#[derive(Debug)]
+pub struct LvolSnapshot {
+    entity_id: Option<String>,
+    parent: Option<String>,
+    group_id: Option<String>,
+    snap_name: Option<String>,
+    lvol: Lvol,
+}
+/// Implement LvolSnapshot Common Function
+impl LvolSnapshot {
+    pub fn create(
+        entity_id: Option<String>,
+        parent: Option<String>,
+        group_id: Option<String>,
+        snap_name: Option<String>,
+        lvol: Lvol,
+    ) -> LvolSnapshot {
+        LvolSnapshot {
+            entity_id,
+            parent,
+            group_id,
+            snap_name,
+            lvol,
+        }
+    }
+}
+/// Trait to give all the common interface of LvolSnapshot.
+trait LvolSnapshotParam {
+    /// Get Group Id of the Snapshot.
+    fn group_id(&self) -> Option<String>;
+
+    /// Get Entity Id of the Snapshot.
+    fn entity_id(&self) -> Option<String>;
+
+    /// Get Parent Id of the Snapshot.
+    fn parent(&self) -> Option<String>;
```

Comment:
This parent id refers to entity_id or txn_id of the parent. you may wan to be specific.  BTW, you may have to search/replace all the group_id into txn_id